src/helpers/LevelJson.h

```diff
@@ -0,0 +1,209 @@
+// bbeck13
+
+#ifndef LEVEL_JSON_H_
+#define LEVEL_JSON_H_
+#include <vector>
+#include <iostream>
+#include <glm/ext.hpp>
+
+#include "MovingPlatform.h"
+#include "DroppingPlatform.h"
+#include "Platform.h"
+#include "json.hpp"
+
+namespace glm {
+void to_json(nlohmann::json& j, const glm::vec3& vec3) {
+  j = nlohmann::json{{"x", vec3.x}, {"y", vec3.y}, {"z", vec3.z}};
+}
+
+void from_json(const nlohmann::json& j, glm::vec3& vec3) {
+  vec3.x = j["x"].get<float>();
+  vec3.y = j["y"].get<float>();
+  vec3.z = j["z"].get<float>();
+}
+}
+
+namespace std {
+void to_json(nlohmann::json& j, const std::vector<glm::vec3>& vec) {
+  for (auto& item : vec) {
+    j.push_back(item);
+  }
+}
+
+void from_json(const nlohmann::json& j, std::vector<glm::vec3>& vec) {
+  for (auto& item : j) {
+    vec.push_back(item);
+  }
+}
+
+// TODO bbeck13 use a template or some shit here
+void to_json(nlohmann::json& j,
+             const std::vector<gameobject::Platform>& items) {
```

Comment:
I realized that I have to enumerate these out since the "level" is a std::vector and it needs to behave very differently than the other vectors of game objects